### PR TITLE
Rename package artifacts from vast to tenzir

### PR DESCRIPTION
**Pull request by tobim**
This change renames user-facing assets from "vast" or a variation thereof to "tenzir".

Automatic migrations of persisted state is implemented on a best-effort basis, meaning that manually created or heavily outdated setups are not supported and require manual intervention.

TODO:
- [x] Adapt Ansible role
- [x] Adapt NixOS module
- [x] Adapt FreeBSD rc file
- [x] Adapt macOS launchd service

### CODE REVIEW

File: vast/CMakeLists.txt [hunk 1]
```diff
@@ -21,22 +29,14 @@ macro (create_tenzir_symlink name)
     CODE "
     execute_process(COMMAND
       \"${CMAKE_COMMAND}\" -E create_symlink \"./tenzir\"
-      \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${name}\")"
+      \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/${name}\")"
```
Comment (dominiklohmann) [inline vast/CMakeLists.txt:32]: This does not look correct to me, shouldn't DESTDIR from ENV affect the install prefix already?

Comment (tobim) [reply vast/CMakeLists.txt:32]: No it doesn't. Look at the `cmake_install.cmake` file in `build/vast/` and you will see.

Comment (dominiklohmann) [reply vast/CMakeLists.txt:32]: If that's the case then all the other instances of `install(CODE …` should be adapted as well.

File: scripts/debian/postinst [hunk 2]
```diff
@@ -3,59 +3,93 @@
 set -e
 
 if [ "$1" = "configure" ]; then
-	mkdir -p /var/lib/vast
-    if ! getent passwd vast >/dev/null; then
+  # Provide an opt-out mechanism for the automatic migration.
```
Comment (dominiklohmann) [inline scripts/debian/postinst:6]: This should go into the changelog entry, I think.

**Review by dominiklohmann — COMMENTED** [review]
Just a first look so far. Will take another look when you ask for a second review.

File: .github/workflows/nix-config-base.json [hunk 3]
```diff
@@ -13,11 +13,11 @@
       ]
     },
     {
-      "name": "vast-ce",
+      "name": "tenzir-ce",
       "static": true,
       "upload-package-to-github": false,
       "package-stores": [
-        "gcs:tenzir-vast/static-binary-builds"
+        "gcs:tenzir-dist-private/static-binary-builds"
```
Comment (dominiklohmann) [inline .github/workflows/nix-config-base.json:20]: This does not yet exist.

File: .github/workflows/nix-manual.yaml [hunk 4]
```diff
@@ -74,11 +74,11 @@ jobs:
               end,
               if ${{ inputs.build-ce }} then
                 {
-                  "name": "vast-ce",
+                  "name": "tenzir-ce",
                   "static": true,
                   "upload-package-to-github": true,
                   "package-stores": [
-                    "gcs:tenzir-vast/static-binary-builds"
+                    "gcs:tenzir-tenzir/static-binary-builds"
```
Comment (dominiklohmann) [inline .github/workflows/nix-manual.yaml:81]: `dist-private`?

Same below in this file.

File: CMakeLists.txt [hunk 5]
```diff
@@ -16,8 +16,8 @@ project(
   HOMEPAGE_URL "https://vast.io"
   LANGUAGES C CXX)
 
-set(VAST_EDITION_NAME
-    "VAST"
+set(TENZIR_EDITION_NAME
+    "tenzir"
```
Comment (dominiklohmann) [inline CMakeLists.txt:20]: Capital T?

File: changelog/next/breaking-changes/3203--vast-to-tenzir-migration.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+The Debian package for tenzir replaces previous VAST installations and attempts
+to migrate existing data from VAST to Tenzir in the process. You can opt-out of
+this migration by creating the file `/var/lib/vast/disable-migration`.
```
Comment (dominiklohmann) [inline changelog/next/breaking-changes/3203--vast-to-tenzir-migration.md:3]:
```suggestion
The Debian package for Tenzir replaces previous VAST installations and attempts
to migrate existing data from VAST to Tenzir in the process. You can opt-out of
this migration by creating the file `/var/lib/vast/disable-migration`.
```

File: tenzir.yaml.example [hunk 7]
```diff
@@ -1,13 +1,13 @@
-# This is an example configuration file for VAST that shows all available
+# This is an example configuration file for Tenzir that shows all available
 # options. Options in angle brackets have their default value determined at
 # runtime.
 
-# Options that concern VAST.
-vast:
+# Options that concern Tenzir.
+tenzir:
```
Comment (dominiklohmann) [inline tenzir.yaml.example:6]: Must be `vast:` until that is also migrated.

File: tenzir.yaml.example [hunk 8]
```diff
@@ -304,7 +305,7 @@ vast:
 
     # Upper bound for the size of a table slice. A value of 0 causes the
     # batch-size to be unbounded, leaving control of batching to the
-    # vast.import.read-timeout option only. This should be a power of 2.
+    # tenzir.import.read-timeout option only. This should be a power of 2.
```
Comment (dominiklohmann) [inline tenzir.yaml.example:308]:
```suggestion
    # vast.import.read-timeout option only. This should be a power of 2.
```

**Review by dominiklohmann — APPROVED** [review]
We peer-reviewed this. A few minor things, then we can merge this. 🚀

File: .github/workflows/nix-config-base.json (hunk 3 above)
Comment (tobim) [reply .github/workflows/nix-config-base.json:20]: I created `tenzir-dist-private` and `tenzir-dist-public` now.

**Pull request merged**